src/backend/aspen/app/views/sample.py

```diff
@@ -416,3 +416,66 @@ def create_sample():
     except Exception as e:
         raise ex.BadRequestException(f"Error encountered when saving data: {e}")
     return jsonify(success=True)
+
+
+@application.route("/api/samples/update/publicids", methods=["POST"])
+@requires_auth
+def update_sample_public_ids():
+    user: User = g.auth_user
+    request_data = request.get_json()
+
+    if not user.system_admin:
+        raise ex.BadRequestException(
+            "user making update request must be a system admin",
+        )
+
+    request_private_ids: list[str] = request_data["private_ids"]
```

Comment:
Why not just accept a map of private -> public ID's ?

Reply:
i don't know why i didn't do that.. lol

Reply:
I think we should update it to accept a map -- less risk of weird off-by-1 issues with two lists

Reply:
added